tower-balance/src/lib.rs

```diff
@@ -151,9 +151,7 @@ where
         debug!("promoting to ready: {}", n);
         // Iterate through the not-ready endpoints from right to left to prevent removals
         // from reordering services in a way that could prevent a service from being polled.
-        for offset in 1..n {
-            let idx = n - offset;
-
+        for idx in (0..n-1).rev() {
```

Comment:
@olix0r I don't think this change is correct --- I stopped seeing the integer overflow in Conduit tests when building with this branch as of commit fb0843496cec45d8e87554f4e58debd8d7f08551, but after pointing back at master as of 777888da7d2f1c8a8058967673ce2c1b70f02adf, I'm getting the overflow again:

```
running 3 tests
test outbound_times_out ... ignored
thread 'support proxy' panicked at 'thread 'attempt to subtract with overflowsupport proxy', ' panicked at '/Users/eliza/.cargo/git/checkouts/tower-b098c32cf5a1bcca/777888d/tower-balance/src/lib.rsattempt to subtract with overflow:', 154/Users/eliza/.cargo/git/checkouts/tower-b098c32cf5a1bcca/777888d/tower-balance/src/lib.rs::24154
:note: Run with `RUST_BACKTRACE=1` for a backtrace.
24
test outbound_reconnects_if_controller_stream_ends ... FAILED
test outbound_asks_controller_api ... FAILED

failures:

---- outbound_reconnects_if_controller_stream_ends stdout ----
	thread 'outbound_reconnects_if_controller_stream_ends' panicked at 'called `Result::unwrap()` on an `Err` value: Error { kind: Inner(Error { kind: Io(Error { repr: Kind(BrokenPipe) }) }) }', src/libcore/result.rs:906:4

---- outbound_asks_controller_api stdout ----
	thread 'outbound_asks_controller_api' panicked at 'called `Result::unwrap()` on an `Err` value: Error { kind: Inner(Error { kind: Io(Error { repr: Kind(BrokenPipe) }) }) }', src/libcore/result.rs:906:4


failures:
    outbound_asks_controller_api
    outbound_reconnects_if_controller_stream_ends

test result: FAILED. 0 passed; 2 failed; 1 ignored; 0 measured; 0 filtered out
```